File: moderate/point_in_circle.c
#include <stdbool.h>
#include <stdio.h>

static bool incircle(float a, float b, float c) {
	return a * a + b * b <= c * c;
}

int main(int argc, char *argv[]) {
	FILE *fp;
	float cx, cy, r, px, py;

	if (argc != 2) {
		printf("Usage: %s [FILE]\n", argv[0]);
		return 1;
	}
	fp = fopen(*++argv, "r");
	while (fscanf(fp, "Center: (%f, %f); Radius: %f; Point: (%f, %f)\n", &cx, &cy, &r, &px, &py) != EOF)
		puts(incircle(cx - px, cy - py, r) ? "true" : "false");
	return 0;
}

File: moderate/point_in_circle.rb
def in_circle(a, b, c)
  a * a + b * b <= c * c
end

p = /\(([^,]+), ([^)]+)\); Radius: ([^;]+); Point: \(([^,]+), ([^)]+)/
File.open(ARGV[0]).each_line do |line|
  s = line.scan(p)[0].map(&:to_f)
  puts in_circle(s[0] - s[3], s[1] - s[4], s[2])
end
